src/sentry/utils/monitors.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+
+import six
+
+from celery.signals import task_prerun, task_postrun
+from django.conf import settings
+from six.moves.urllib.parse import urlparse
+from time import time
+
+from sentry.net.http import SafeSession
+from sentry.utils.sdk import configure_scope
+
+
+def get_api_root_from_dsn(dsn):
+    if not dsn:
+        return
+    parsed = urlparse(dsn)
+    return u'{}://{}:{}'.format(parsed.scheme, parsed.hostname, parsed.port)
+
+
+SENTRY_DSN = settings.SENTRY_MONITOR_DSN
+
+API_ROOT = settings.SENTRY_MONITOR_API_ROOT or get_api_root_from_dsn(SENTRY_DSN)
+
+
+def connect(app):
+    task_prerun.connect(report_monitor_begin, weak=False)
+    task_postrun.connect(report_monitor_complete, weak=False)
+
+    # XXX(dcramer): Celery docs suggest it should be app.conf.beat_schedule, which
+    # was likely a change in 4.x. This code is intended to support "any celery" and be
+    # adopted into sentry-sdk core, thus we support it here.
+    schedule = app.conf.beat_schedule if hasattr(
+        app.conf, 'beat_schedule') else app.conf['CELERYBEAT_SCHEDULE']
+    for schedule_name, monitor_id in six.iteritems(settings.SENTRY_CELERYBEAT_MONITORS):
+        schedule[schedule_name].setdefault(
+            'options',
+            {}).setdefault(
+            'headers',
+            {}).setdefault(
+            'X-Sentry-Monitor',
+            monitor_id)
+    # before_task_publish.connect(augment)
+
+
+# Celery signals fail to propagate if they error and we dont want to break things with our
+# instrumentation
```

Comment:
This doesn't seem to be a problem with celery 4.2.1 (maybe earlier versions too), but I can repro this with our current celery. Not that it's of any help currently, but I would add this info in case we decide to upgrade.